lib/pbench/server/api/resources/query_apis/__init__.py

```diff
@@ -389,61 +374,74 @@ def _call(self, method: Callable, params: ApiParams, context: ApiContext) -> JSO
             raise APIAbort(e.http_status, str(e))
         except KeyError as e:
             raise APIInternalError(f"problem in preprocess, missing {e}") from e
+
+        url = None
         try:
             # prepare payload for Elasticsearch query
             es_request = self.assemble(params, context)
-            path = es_request.get("path")
-            url = urljoin(self.es_url, path)
-            current_app.logger.info(
-                "ASSEMBLE returned URL {!r}, {!r}",
-                url,
-                es_request.get("kwargs").get("json"),
-            )
+            if not es_request:
+                current_app.logger.info("ASSEMBLE disabled Elasticsearch call")
+            else:
+                path = es_request.get("path")
+                url = urljoin(self.es_url, path)
+                current_app.logger.info(
+                    "ASSEMBLE returned URL {!r}, {!r}",
+                    url,
+                    es_request.get("kwargs").get("json"),
+                )
```

Comment:
I didn't change this code, aside from adding indentation to allow the "Elasticsearch-based" delete and update to work when there are no indexed documents. And while it did occur to me that this might not be a bad time to downgrade to `DEBUG`, if I do that in this PR it'll be *after* I've finished the unit tests.